src/keystore/config.ts

```diff
@@ -4,6 +4,7 @@ import { KeyStore, CryptoSystem } from 'keystore-idb/types'
 const KEYSTORE_CFG = { type: CryptoSystem.RSA }
 
 
+// Is the keystore a singleton?
```

Comment:
This pattern is in a few places. Are we trying to explicitly limit the number of keystores?

Reply:
You only need one keystore for the sdk, and this lets you access that keystore from anywhere in the code without passing it around as a parameter to everything.

I don't like singletons very much (even tho I use them twice in the sdk :sweat_smile:), but it seemed like a good fit here. Open to alternatives!

Reply:
I definitely get that this is convenient. Purely for exploration, would there be a problem with something like:

```ts
const {keystore, filesystem} = Fission.setup({...})
```

Where the filesystem also has a link to that keystore on setup. This is more flexible (say you want to load multiple filesystems, or keep two applications totally separate). But maybe the tradeoff for convenience is too high? Not sure, very much looking for your thoughts here.

Reply:
:thinking: :thinking: Looking back at the ts-sdk, we really don't even use that singleton very much (especially in the filesystem, which is all symm keys). 

In that case would we still have the singleton? one-off methods such as `core.did()` need access to a canonical keystore for determining the user's public key